liquidctl/config/validation.py

```diff
@@ -0,0 +1,157 @@
+import logging
```

Comment:
General comment: this looks hard to maintain in the long run.

Have you consider a generic solutions, perhaps relying on each driver exposing some additional information?

Reply:
I too am not super happy with how I am approaching this, it feels horrendous. But I also think this general structure would not be that hard to maintain because each function is responsible for checking a single block and then it delegates the sub block to a different function. 

But Ill look  into other config file parsers to see how they handle similar cases

Reply:
I don't think the parsing is the issue, but rather how we map this to the actual devices and their capabilities.  That's the part that I'm worried might be hard to maintain.